Lib/xmlrpc/server.py

```diff
@@ -386,30 +386,31 @@ def _dispatch(self, method, params):
         not be called.
         """
 
-        func = None
         try:
             # check to see if a matching function has been registered
             func = self.funcs[method]
         except KeyError:
-            if self.instance is not None:
+            try:
                 # check for a _dispatch method
-                if hasattr(self.instance, '_dispatch'):
-                    return self.instance._dispatch(method, params)
-                else:
-                    # call instance method directly
-                    try:
-                        func = resolve_dotted_attribute(
-                            self.instance,
-                            method,
-                            self.allow_dotted_names
-                            )
-                    except AttributeError:
-                        pass
+                func = self.instance._dispatch
+            except AttributeError:
+                # call instance method directly
+                try:
+                    func = resolve_dotted_attribute(
```

Comment:
`None` is handled special in current code. Existing code can set an attribute to `None` for disabling an inherited method. A user get a more informative exception *'method "{methodname}" is not supported'* rather than *"'NoneType' object is not callable"*. I don't know whether this is used in real code, but there is a risk of breaking existing code.